fdasd: Add non-interactive script friendly interface

vojtechtrefny opened the pull request:
This is just a work in progress patch, mostly giving you a chance to tell me you are not interested in this and/or suggest a better approach.
The idea is to provide a script friendly interface similar to `parted --script`.  Calling this can look for example like

```
# fdasd --script "d 1 a 2 +100m w" /dev/dasdc
```
to remove an existing partition and add a new 100 MiB one. 

I am trying to reuse as much of the existing code as possible so this hopefully shouldn't create more maintenance work in the future.
This is obviously missing other commands, documentation and needs some more testing etc, as I said, still work in progress and mostly looking for some early feedback now.

steffen-eiden commented:
Hey, this sounds good.I had a quick glance looks good.

I know this is just an early draft, but for the real deal please split up the patch into multiple and first do all the refactoring (e.g. move stuff to separate functions) ans then the actual feature addition.|

I am not soo fluent with the dasd tools but you may be able to reuse the CLI parser to parse you script.
(under the assumption that the script mode cannot do more than the interactive)

@hoeppnerj  may provide more information and/or feedback

vojtechtrefny committed:
fdasd: Split parsing size to a separate helper function

Signed-off-by: Vojtech Trefny <vtrefny@redhat.com>

vojtechtrefny committed:
fdasd: Split parsing size to a separate helper function

Signed-off-by: Vojtech Trefny <vtrefny@redhat.com>

vojtechtrefny committed:
fdasd: Split fdasd_add_partition into two functions

This splits the add partition functionality into two functions so
the part that doesn't require user input can be reused.

Signed-off-by: Vojtech Trefny <vtrefny@redhat.com>

vojtechtrefny committed:
fdasd: Split fdasd_add_partition into two functions

This splits the add partition functionality into two functions so
the part that doesn't require user input can be reused.

Signed-off-by: Vojtech Trefny <vtrefny@redhat.com>

vojtechtrefny committed:
fdasd: Split fdasd_remove_partition into two functions

This splits the remove partition functionality into two functions
so the part that doesn't require user input can be reused.

Signed-off-by: Vojtech Trefny <vtrefny@redhat.com>

vojtechtrefny committed:
fdasd: Split fdasd_remove_partition into two functions

This splits the remove partition functionality into two functions
so the part that doesn't require user input can be reused.

Signed-off-by: Vojtech Trefny <vtrefny@redhat.com>

vojtechtrefny committed:
fdasd: Add non-interactive script friendly interface

This adds a scripting interface similar to how 'parted --script'
works.

Signed-off-by: Vojtech Trefny <vtrefny@redhat.com>

vojtechtrefny committed:
fdasd: Add non-interactive script friendly interface

This adds a scripting interface similar to how 'parted --script'
works.

Signed-off-by: Vojtech Trefny <vtrefny@redhat.com>

vojtechtrefny committed:
fdasd: Add the new -S/--script option to man page

Signed-off-by: Vojtech Trefny <vtrefny@redhat.com>

vojtechtrefny committed:
fdasd: Add the new -S/--script option to man page

Signed-off-by: Vojtech Trefny <vtrefny@redhat.com>

vojtechtrefny committed:
fdasd: Add a simple test for the new script mode

Signed-off-by: Vojtech Trefny <vtrefny@redhat.com>

vojtechtrefny committed:
fdasd: Add a simple test for the new script mode

Signed-off-by: Vojtech Trefny <vtrefny@redhat.com>
